File: api/src/endpoint/user/update_current_user.rs
use serde::{Deserialize, Serialize};

use crate::{endpoint::Endpoint, resource::user::UserResource};

pub struct UpdateCurrentUser;

impl Endpoint for UpdateCurrentUser {
    const PATH: &'static str = "/user";
    const METHOD: http::Method = http::Method::PATCH;

    type Request = UpdateCurrentUserRequest;
    type Response = UpdateCurrentUserResponse;
}

#[derive(ApiRequest, Debug, Serialize, Deserialize)]
pub struct UpdateCurrentUserRequest {
    pub name: String,
}

pub type UpdateCurrentUserResponse = UserResource;

File: gitdot-api-ts/src/endpoint/user/update-current-user.ts
import { z } from "zod";
import { UserResource } from "../../resource";
import type { Endpoint } from "../endpoint";

export const UpdateCurrentUserRequest = z.object({
  name: z.string(),
});
export type UpdateCurrentUserRequest = z.infer<typeof UpdateCurrentUserRequest>;

export const UpdateCurrentUserResponse = UserResource;
export type UpdateCurrentUserResponse = z.infer<
  typeof UpdateCurrentUserResponse
>;

export const UpdateCurrentUser = {
  path: "/user",
  method: "PATCH",
  request: UpdateCurrentUserRequest,
  response: UpdateCurrentUserResponse,
} satisfies Endpoint;
export type UpdateCurrentUser = typeof UpdateCurrentUser;
